app/src/main/java/com/vero/tasky/auth/domain/usecase/password/ValidatePasswordUseCase.kt

```diff
@@ -4,10 +4,10 @@ import java.util.regex.Pattern
 
 class ValidatePasswordUseCase {
 
-    private val pattern = Pattern.compile("^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)")
+    private val pattern = Pattern.compile("^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d).+\$")
```

Comment:
I personally am not a big fan of Regex, due to this thing (You had to modify it for tests and so on)

And unless you test it 100%, you aren't fully sure what can happen

I prefer having a bunch of if statements on the code, to check specific parts

Ex:

if (password.length < 9) return Too Short
if (!password.any {it.isLowerCase()}) return No Lower Case
if (!password.any {it.isUpperCase()}) return No Upper Case

And so on

Also, if you read Tasky's requirements. The password NEEDS to have at least:
1 upper case
1 lower case
1 number

Which you aren't doing by applying the Regex



For email, on the other hand. It´s okay to use Regex (Android pattern´s Email regex). Because an email is either valid or invalid.

Reply:
At least for things where you can easily write a bunch of if statements like validating a password. For complex patterns regex is totally fine and legit